packages/venia-concept/src/components/Head/headProvider.js

```diff
@@ -0,0 +1,21 @@
+import React, { useEffect } from 'react';
+import { HeadProvider as _HeadProvider } from 'react-head';
+
+const filterOutTitleTags = titleTags => {
+    titleTags
+        .filter(titleNode => !titleNode.dataset.synthetictag)
+        .forEach(titleTag => {
+            if (titleTag) titleTag.parentNode.removeChild(titleTag);
+        });
+};
+
+const HeadProvider = props => {
+    useEffect(() => {
+        const titleTagsToRemove = document.getElementsByTagName('title');
+        filterOutTitleTags([...titleTagsToRemove]);
+    }, []);
```

Comment:
I'm still hoping we can simplify this 😄 

Because we're passing an empty array as the second argument to `useEffect`, it will run on mount and unmount only:

> If you want to run an effect and clean it up only once (on mount and unmount), you can pass an empty array ([]) as a second argument. This tells React that your effect doesn’t depend on any values from props or state, so it never needs to re-run. This isn’t handled as a special case — it follows directly from how the dependencies array always works.

Since our `Title` component doesn't do anything unless it's wrapped in this `HeadProvider` component, there can never be a case where the `Title` component has inserted a `<title>` that the `HeadProvider` wants to skip removing, right?

If I'm understanding all that correctly, then there is no need for the synthetic tag data attribute, and therefore no need to filter for them here - just removing all `<title>` tags should be sufficient 👍

Reply:
You are right. But that was added in place to handle the server-rendered output. If the server-rendered output already has a `synthetictag` on it, we don't want to delete it.

But after talking to you, I have realized that is a bleak use-case. We can probably get rid of the filtering for now and add this logic when needed.